java-checks/src/main/java/org/sonar/java/checks/RecordDuplicatedGetterCheck.java

```diff
@@ -0,0 +1,150 @@
+    if (isDirectCallToAccessor(getter, component) || isPojoGetter(getter, component)) {
+      return;
+    }
+    Optional<MethodTree> accessor = findDeclaredMethod(recordSymbol, component.name());
+    if (!accessor.isPresent()) {
+      reportIssue(getter.simpleName(), issueMessage(getter, component));
+    } else {
+      MethodTree accessorMethod = accessor.get();
+      if (!SyntacticEquivalence.areEquivalent(accessorMethod.block(), getter.block())) {
+        reportIssue(getter.simpleName(), issueMessage(getter, component));
+      }
+    }
+  }
+
+  private static boolean isPojoGetter(MethodTree method, Symbol.VariableSymbol component) {
+    return singleReturnStatementExpression(method).filter(expr -> isComponent(expr, component)).isPresent();
+  }
+
+  private static boolean isDirectCallToAccessor(MethodTree getter, Symbol.VariableSymbol component) {
+    return singleReturnStatementExpression(getter).filter(expr -> isAccessorInvocation(expr, component)).isPresent();
+  }
+
+  private static Optional<ExpressionTree> singleReturnStatementExpression(MethodTree method) {
+    return Optional.ofNullable(method.block())
+      .filter(b -> b.body().size() == 1)
+      .map(b -> b.body().get(0))
+      .filter(s -> s.is(Tree.Kind.RETURN_STATEMENT))
+      .map(ReturnStatementTree.class::cast)
+      .map(ReturnStatementTree::expression)
+      .filter(Objects::nonNull);
+  }
+
+  private static boolean isComponent(ExpressionTree expression, Symbol.VariableSymbol component) {
+    if (expression.is(Tree.Kind.IDENTIFIER)) {
+      return component.equals(((IdentifierTree) expression).symbol());
+    }
+    if (expression.is(Tree.Kind.MEMBER_SELECT)) {
+      MemberSelectExpressionTree mset = (MemberSelectExpressionTree) expression;
+      return ExpressionUtils.isThis(mset.expression())
+        && isComponent(mset.identifier(), component);
+    }
+    return false;
+  }
+
+  private static boolean isAccessorInvocation(ExpressionTree expression, Symbol.VariableSymbol component) {
+    if (!expression.is(Tree.Kind.METHOD_INVOCATION)) {
+      return false;
+    }
+    MethodInvocationTree mit = (MethodInvocationTree) expression;
+    Symbol methodSymbol = mit.symbol();
+    return mit.arguments().isEmpty()
+      && component.name().equals(methodSymbol.name())
+      && component.owner().equals(methodSymbol.owner());
+  }
+
+  private static String issueMessage(MethodTree getter, Symbol.VariableSymbol component) {
+    return String.format("Remove this getter '%s()' from record and override an existing one '%s()'.", getter.simpleName().name(), component.name());
+  }
+
+  private static List<Symbol.VariableSymbol> recordComponents(Symbol.TypeSymbol recordSymbol) {
+    return recordSymbol
+      .memberSymbols()
+      .stream()
+      .filter(Symbol::isVariableSymbol)
+      .map(Symbol.VariableSymbol.class::cast)
+      .filter(component -> !"this".equals(component.name()))
```

Comment:
I don't understand what this check does. Why/how would a member's name be `this`? Does that happen in case of inner classes? If so, nested records are always static, so they'll never hold a reference to the outer `this`.

Reply:
My bad. In our previous semantic (before ECJ), we were artificially introducing a "this" symbol in the list of member symbols. This is not done, nor required anymore.